source/common/config/config_provider_impl.h

```diff
@@ -298,19 +388,58 @@ class MutableConfigProviderImplBase : public ConfigProvider {
 
     ConfigProvider::ConfigConstSharedPtr config_;
   };
+};
 
-  ConfigSubscriptionInstanceBaseSharedPtr subscription_;
-  ThreadLocal::SlotPtr tls_;
+/**
+ * Provides common mutable (dynamic) config provider functionality required by
+ * ConfigProvider::ApiType::Delta DS APIs.
+ */
+class DeltaMutableConfigProviderBase : public MutableConfigProviderCommonBase {
+public:
+  // Envoy::Config::ConfigProvider
+  // This promotes getConfig() to public so that internal uses can avoid an unnecessary dynamic_cast
+  // in the public API (ConfigProvider::config<T>()).
+  ConfigConstSharedPtr getConfig() const override { NOT_IMPLEMENTED_GCOVR_EXCL_LINE; }
+
+  /**
+   * Non-const overload for use within the framework.
+   * @return ConfigSharedPtr the config implementation associated with the provider.
+   */
+  virtual ConfigSharedPtr getConfig() PURE;
+
+  /**
+   * Propagates a delta config update to all workers.
```

Comment:
Can you explain how subtractive updates work, e.g. removing a resource in a delta. I claimed in one of my other comments that this isn't supported, but I think that might not be the case. Since the tests below don't demo this, it would be interested to hear how it would work.